src/deye_daemon.py

```diff
@@ -117,8 +114,8 @@ def __remove_duplicated_reg_ranges(self, reg_ranges: list[SensorRegisterRange])
                 result.append(reg_range)
         return result
 
-    def __new_event(self, events: list[DeyeEvent]) -> bool:
-        """Check if the received event data has changed compared
+    def __is_device_observation_changed(self, events: DeyeEventList) -> bool:
+        """Check if the received event observations have changed compared to last published
```

Comment:
:+1:

Reply:
Any reason why the method name should start with  `__` ? I don't see a risk of name clashes here.

Using the class as type hint within the class is not possible because it's not defined yet (causes linting and runtime error). I add string based type hints as per [PEP 484](https://peps.python.org/pep-0484/#forward-references).